sql/src/main/java/org/apache/druid/sql/avatica/DruidConnection.java

```diff
@@ -52,26 +52,30 @@
   private final ImmutableMap<String, Object> context;
   private final AtomicInteger statementCounter = new AtomicInteger();
   private final AtomicReference<Future<?>> timeoutFuture = new AtomicReference<>();
-
-  @GuardedBy("statements")
   private final Map<Integer, DruidStatement> statements;
```

Comment:
IMO, things would be more clear if you changed the type of `statements` to a ConcurrentMap and added one more comment here about how it should be accessed. The variable is sometimes accessed under lock and sometimes not, and that can be tough for readers to figure out.

```java
// Typically synchronized by connectionLock, except in one case: the onClose function passed
// into DruidStatements contained by the map.
private final ConcurrentMap<Integer, DruidStatement> statements;
```

Reply:
OK, the ConcurrentMap type will be more clear, I'll change it.